src/components/button/index.js

```diff
@@ -175,8 +175,8 @@ export default class Button extends BaseComponent {
     }
   }
 
-  componentDidUpdate() {
-    if (this.props.animateLayout) {
+  componentWillReceiveProps(nextProps) {
+    if (!_.isEqual(nextProps, this.props) && this.props.animateLayout) {
```

Comment:
You can still keep using `componentDidUpdate`
`componentDidUpdate` param is `prevProps` which you can compare to `this.props`.
I rather keep using `componentDidUpdate` since it considered best practice